File: service/auth-service/src/main/java/com/simp/service/auth/service/AuthServiceImpl.java
package com.simp.service.auth.service;

import com.simp.service.shared.domain.service.AuthService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthServiceImpl implements AuthService {
    @Override
    public String getGreeting(String name) {
        return "Hello, %s [from auth-service]".formatted(name);
    }
}

File: service/auth-service/src/main/kotlin/com/simp/service/auth/service/AuthServiceImpl.kt
package com.simp.service.auth.service

import com.simp.service.shared.domain.service.AuthService
import org.springframework.stereotype.Service

@Service
class AuthServiceImpl : AuthService {
    override fun getGreeting(name: String): String =
        "Hello, $name [from auth-service]"
}